src/Cache/FileCache.php

```diff
@@ -15,16 +15,16 @@ class FileCache implements CacheInterface
 
     public function __construct(string $dir)
     {
-        if (!is_dir($dir)) {
-            throw new \InvalidArgumentException(sprintf('The directory "%s" does not exist.', $dir));
+        if (!is_dir($dir) && false === @mkdir($dir, 0775, true)) {
```

Comment:
why not `777` ?

Reply:
Main reason: I copied it from the previous PR. But I'm also not sure on why the cache should be world readable/writable, so maybe `770` would actually be better... But setting it to `777` would probably be best. Looks like all the other PHP libraries do that as well.